### PR TITLE
Use latest version of fourmolu possible

### PR DESCRIPTION
**Pull request by brandonchinn178**
I don't use Nix, so I can't update any of the Nix files

Broken out from https://github.com/haskell/haskell-language-server/pull/3762

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -30,7 +30,7 @@ packages:
          ./plugins/hls-qualify-imported-names-plugin
          ./plugins/hls-code-range-plugin
          ./plugins/hls-change-type-signature-plugin
-         ./plugins/hls-stan-plugin
+         -- ./plugins/hls-stan-plugin
```
Comment (brandonchinn178) [inline cabal.project:33]: hls-stan-plugin is breaking my builds: https://github.com/haskell/haskell-language-server/actions/runs/5893488060/job/15985146763?pr=3764

https://github.com/haskell/haskell-language-server/issues/3701

**Comment by joyfulmantis** [comment]
Updating nix isn't important, but we might want to get one more release out with stan.

**Review by joyfulmantis — APPROVED** [review]
LGTM

**Comment by brandonchinn178** [comment]
@joyfulmantis can you help with the failures? I'm not sure how they're related to my changes

**Comment by joyfulmantis** [comment]
Yeah they're wierd, especially the 8.10 build seems like it's being killed because out of memory or something. I can restart them, but possibly CI issue, possibly your changes are bringing some wierd problem to the front.

**Comment by brandonchinn178** [comment]
@michaelpj Can you help with these CI failures?

**Review by michaelpj — APPROVED** [review]

**Pull request merged**